File: Tree/563.cpp
class Solution {
public:
    int res = 0;
    int findTilt(TreeNode* root) {
        dfs(root);
        return res;
    }

    int dfs(TreeNode* root) {
        if (!root) return 0;
        int left = dfs(root->left);
        int right = dfs(root->right);
        res = res + abs(left - right);

        return root->val + left + right;
    }

};

File: Tree/563.py
class Solution:
    def findTilt(self, root: Optional[TreeNode]) -> int:
        self.ans = 0

        def dfs(node):
            if not node: return 0
            left, right = dfs(node.left), dfs(node.right)
            self.ans += abs(left - right)
            return node.val + left + right
        
        dfs(root)

        return self.ans
